cookbook/entities/acl.rst

```diff
@@ -0,0 +1,336 @@
+        :doc:`/reference/annotation/acl_ancestor`, and :doc:`ACL YAML format </reference/format/acl>`
+        reference.
+
+.. _coobook-entities-acl-check:
+
+Perform Access Checks
+---------------------
+
+Once you have configured the ACLs you can protect all parts of your application. Anywhere in your
+PHP code you can use the ``isGranted()`` method of the ``oro_security.security_facade`` service
+(which is an instance of the :class:`Oro\\Bundle\\SecurityBundle\\SecurityFacade` class):
+
+.. code-block:: php
+    :linenos:
+
+    $securityFacade = $this->container->get('oro_security.security_facade');
+
+    if ($securityFacade->isGranted('app_task_create')) {
+        // do something when the user is granted permissions for the app_static_pages ACL
+    }
+
+However, there are ways to configure these access checks in your YAML config files or via
+annotations in several places which make their usage a lot easier:
+
+Hiding Menu Items
+~~~~~~~~~~~~~~~~~
+
+Use the ``aclResourceId`` option to hide navigation items from users who are not granted to access
+the action being linked. The value of this option is the name of the ACL to check for:
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/AppBunde/Resources/config/navigation.yml
+    oro_menu_config:
+        items:
+            task_list:
+                label: Tasks
+                route: app_task_index
+                aclResourceId: app_task_view
+
+Protecting Controllers Refering to Existing ACLs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As :ref:`shown above <cookbook-entity-acl-controller>` you can define new ACLs and protect your
+controllers with them in a single step using the ``@Acl`` annotation. However, you can also refer
+to an existing access control list using the ``@AclAncestor`` annotation:
+
+.. code-block:: php
+    :linenos:
+
+    // src/AppBunde/Controller/TaskController.php
+    namespace AppBundle\Controller;
+
+    use AppBunde\Entity\Task;
+    use Oro\Bundle\SecurityBundle\Annotation\AclAncestor;
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+    class TaskController extends Controller
+    {
+        /**
+         * @AclAncestor("app_task_view")
+         */
+        public function viewAction(Task $task)
+        {
+            // ...
+        }
+
+        // ...
+    }
+
+Show Parts of Templates Based on Permissions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Inside your templates you can use the ``resource_granted()`` Twig function to check for certain
+permissions to hide parts of your views for users who do not have the required permissions:
+
+.. code-block:: html+jinja
+    :linenos:
+
+    {# src/AppBundle/Resources/views/Task/index.html.twig #}
+    {% extends 'OroUIBundle:actions:index.html.twig' %}
+
+    {% set gridName = 'app-tasks-grid' %}
+    {% set pageTitle = 'Task' %}
+
+    {% block navButtons %}
+        {% if resource_granted('app_task_create') %}
```

Comment:
I think you can delete extends, set, block tags from this example. as the result, the example will be more readable.

And please add more examples with different data (entity, class name, etc)